OIA-29: Create Maven archetype for .kar plugin

j-white opened the pull request:
Adds an archetype that can be used to quickly bootstrap new plugin development.

j-white commented:
This can be tested by building the tree locally and then executing the archetype generation command in the README. This should generate a new plugin that can be cleanly compiled and loaded in OpenNMS.

patrick-schweizer reviewed (commented):
I tested the generation and usage of the plugin in OpenNMS. It seems to work well in general. This is idea great - it saves a lot of setup work for new plugins! :-)

A few comments:

- I tried to follow the README and tried out the features described there. The following command didn't work for me:
![image](https://user-images.githubusercontent.com/1060134/92285187-50d20600-eec9-11ea-9f40-387d573decb5.png)
- The created plug in comes with some example code. I am wondering if it will help people to give some more explanation on what classes the plugin contains and what they do? Something like:

```
The generated plugin contains the following code examples to demonstrate how to create a:
- Karaf Shell Command (see StatsCommand)
- Rest Endpoint (see WebhookHandler)
- ...
```
and then some explanation in the classes with links to our documentation.

patrick-schweizer commented on archetypes/example-kar-plugin/src/main/resources/archetype-resources/plugin/src/main/java/AlarmForwarder.java:
```diff
@@ -0,0 +1,147 @@
+#set( $symbol_dollar = '$' )
+#set( $symbol_escape = '\' )
+/*******************************************************************************
+ * This file is part of OpenNMS(R).
```
Does the OpenNMS License apply for generated plug ins? i assume when we write yes, for external users - I don't know?

j-white replied:
Good point - We need to investigate this topic further - Will follow up in OIA-30

patrick-schweizer commented on archetypes/example-kar-plugin/src/main/resources/archetype-resources/README.md:
```diff
@@ -0,0 +1,33 @@
+* opennms-${pluginId}:stats
+* opennms-${pluginId}:push-topology
+
+You can also access the REST endpoint mounted by the plugin at `http://nms:8980/opennms/rest/${pluginId}/ping`
```
I wonder if it shouldn't be _localhost_ instead of _nms_? Most developers will run the system locally?

j-white replied:
Done.

j-white commented:
Added a bit more docs and removed the copyright headers.

The connection error when forwarding the topologies is expected to happen since there's nothing to forward to.

patrick-schweizer reviewed (approved):
:+1:
